### ISSUE TITLE
Apparent memory leak introduced with v2.1.7

### ISSUE THREAD

**Issue by ginkel** [containous/traefik#6520]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

We have deployed Traefik v2.1 on a couple of nodes, limited to 128 MB memory each. This setup has been running stable ever since Trafik 2.x was released.

Last night our Traefik deployments have been automatically updated from v2.1.6 to v2.1.7.

Since then the instances are regularly OOM-killed. The following Grafana chart shows the difference in behavior (old: constant memory usage after a ramp-up, new: restart cycles due to OOMs):

![image](https://user-images.githubusercontent.com/174129/77042116-6c992c80-69bb-11ea-917f-c1faf4c5dc36.png)

![image](https://user-images.githubusercontent.com/174129/77042217-96eaea00-69bb-11ea-97d9-6377d8f8ff46.png)

### What did you expect to see?

Traefik continues running as before.

### What did you see instead?

Traefik is periodically killed by the Linux kernel due to an OOM (when hitting the 128 MB memory limit).

The issue can be reproduced more quickly by firing plenty of requests against the instance, e.g., using `ab`:

```
ab -n 100000 -c 10 https://<fqdn>/
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
$ docker exec -it traefik traefik version
Version:      2.1.7
Codename:     cantal
Go version:   go1.13.8
Built:        2020-03-18T16:43:27Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Linux amd64, running on Ubuntu 18.04 w/ Docker 19.03.8

```yaml
accessLog:
    filePath: /log/access.log
api:
    dashboard: true
    debug: false
entryPoints:
    http:
        address: :80
    https:
        address: :443
    traefik:
        address: :81
log:
    format: json
    level: INFO
metrics:
    prometheus:
        entryPoint: traefik
providers:
    docker:
        exposedByDefault: false
    file:
        directory: /etc/traefik/rules
        watch: true
serversTransport:
    insecureSkipVerify: true
```

**Comment by ldez** [comment]
Thank you @ginkel :+1: 

We fixed that in #6522, and we will create a release today.

**Comment by traefiker** [comment]
Closed by #6522.
